src/emitter.ts

```diff
@@ -320,6 +320,18 @@ export function emitWebIdl(webidl: Browser.WebIdl, flavor: Flavor, iterator: boo
         };
     }
 
+    function convertDomTypeToTsTypeInUnion(obj: Browser.Typed): { name: string; nullable: boolean } {
```

Comment:
I'm not sure why this is needed, what happens without this?

Reply:
[See this comment.](https://github.com/microsoft/TSJS-lib-generator/pull/893#discussion_r487332968) In most cases, we want to emit `undefined` as `void`, which is handled by 8dc0f62c0175e1c2ef29ab9bc49a8b5102a625a4. However, if this is a type inside of a union type, we want to special-case it and turn it back into `undefined`.

Perhaps it's a bit confusing that we first transform `undefined` to `void` and then back to `undefined`? Do you want me to try to flip the logic around: use `undefined` by default (i.e. add it to `sameTypes` in [helpers.js](https://github.com/microsoft/TSJS-lib-generator/blob/8dc0f62c0175e1c2ef29ab9bc49a8b5102a625a4/src/helpers.ts#L8)), but add a special case for when it does *not* appear in a union type?

Reply:
> Do you want me to try to flip the logic around: use `undefined` by default (i.e. add it to `sameTypes` in [helpers.js](https://github.com/microsoft/TSJS-lib-generator/blob/8dc0f62c0175e1c2ef29ab9bc49a8b5102a625a4/src/helpers.ts#L8)), but add a special case for when it does _not_ appear in a union type?

Hmm yes, that sounds more intuitive. 👍